src/main/java/seedu/pill/util/ItemMap.java

```diff
@@ -0,0 +1,335 @@
+        Optional<LocalDate> expiryDate = updatedItem.getExpiryDate();
+
+        assert name != null && !name.trim().isEmpty() : "Item name cannot be null or empty";
+        assert quantity > 0 : "Quantity must be positive";
+
+        if (name == null || name.trim().isEmpty() || quantity <= 0) {
+            LOGGER.warning("Attempt to edit item with invalid parameters: name=" + name + ", quantity=" + quantity);
+            System.out.println("Invalid item name or quantity.");
+            return;
+        }
+
+        TreeSet<Item> itemSet = items.get(name);
+        boolean isUpdated = false;
+        if (itemSet != null) {
+            for (Item item : itemSet) {
+                if (item.getExpiryDate().equals(expiryDate)) {
+                    item.setQuantity(quantity);
+                    isUpdated = true;
+                }
+            }
+            if (isUpdated) {
+                LOGGER.info("Edited item: " + updatedItem);
+                System.out.println("Edited item: " + updatedItem);
+            } else {
+                LOGGER.warning("Attempt to edit non-existent item: " + updatedItem);
+                System.out.println("Item not found: " + updatedItem);
+            }
+        } else {
+            LOGGER.warning("Attempt to edit non-existent item: " + name);
+            System.out.println("Item not found: " + name);
+        }
+    }
+
+    /**
+     * Lists all the items in the inventory.
+     */
+    public void listItems() {
+        if (items.isEmpty()) {
+            LOGGER.info("Attempted to list items, but inventory is empty");
+            System.out.println("The inventory is empty.");
+            return;
+        }
+        LOGGER.info("Listing all items in inventory");
+        System.out.println("Listing all items:");
+        int index = 1;
+        for (Map.Entry<String, TreeSet<Item>> entry : items.entrySet()) {
+            TreeSet<Item> itemSet = entry.getValue();
+            for (Item item : itemSet) {
+                System.out.println(index + ". " + item.toString());
+                index++;
+            }
+        }
+    }
+
+    /**
+     * Lists all the items in the inventory for restock command, given a threshold value.
+     * Prints all items with quantity strictly less than threshold.
+     *
+     * @param threshold The minimum number of items before it is deemed to require replenishment.
+     */
+    public void listItemsToRestock(int threshold){
+        try {
+            if (items.isEmpty()) {
+                LOGGER.info("Attempted to list items, but inventory is empty");
+                return;
+            }
+
+            if (threshold < 0) {
+                throw new PillException(ExceptionMessages.INVALID_QUANTITY);
+            }
+
+            List<Item> filteredItems = items.values().stream()
+                    .flatMap(TreeSet::stream)
+                    .filter(item -> item.getQuantity() <= threshold)
+                    .toList();
```

Comment:
Maybe you can look into refactoring this to make it more readable and less complicated?